File: src/main/java/kr/co/lunatalk/domain/payment/event/PaymentCompletedEvent.java
package kr.co.lunatalk.domain.payment.event;

import java.util.List;

public record PaymentCompletedEvent(
	String orderNumber,
	Long orderId,
	Long totalAmount,
	String memberEmail,
	List<PaymentOrderItem> items
) {
	public record PaymentOrderItem(
		Long productId,
		String productName,
		Integer quantity,
		Long price
	) {
	}
}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/payment/event/PaymentCompletedEvent.kt
package kr.co.lunatalk.domain.payment.event

data class PaymentCompletedEvent(
    val orderNumber: String,
    val orderId: Long,
    val totalAmount: Long,
    val memberEmail: String,
    val items: List<PaymentOrderItem>
) {
    data class PaymentOrderItem(
        val productId: Long,
        val productName: String,
        val quantity: Int,
        val price: Long
    )
}
